GenderPayGap.WebUI/ScheduledJobs/HangfireConfiguration/HangfireAuthorisationFilter.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Linq;
+using System.Security.Claims;
+using Hangfire.Dashboard;
+
+namespace GenderPayGap.WebUI.ScheduledJobs.HangfireConfiguration
+{
+    public class HangfireAuthorisationFilter : IDashboardAuthorizationFilter
+    {
+        public bool Authorize(DashboardContext context)
+        {
+            var httpContext = context.GetHttpContext();
+
+            if (!httpContext.User.Identity.IsAuthenticated)
+            {
+                return false;
+            }
+
+            var claims = httpContext.User.Claims;
+            Claim roleClaim = claims.FirstOrDefault(c => c.Type == ClaimTypes.Role);
+
+            bool isAdministrator = roleClaim != null && roleClaim.Value == "GPGadmin";
+
+            return isAdministrator;

```

Comment:
Are you able to explain a bit about what is going on here? Why would Hangfire need to know about Users?

Reply:
Hangfire has a dashboard (/admin/hangfire).
This code makes sure that only our GPG Admin users are able to access it.

Reply:
That's cool, thanks for the explanation!